esmvalcore/config/_validated_config.py

```diff
@@ -133,3 +134,26 @@ def copy(self):
     def clear(self):
         """Clear Config."""
         self._mapping.clear()
+
+    @contextmanager
+    def context(
+        self,
+        mapping: Mapping | None = None,
+        **kwargs: Any,
+    ) -> Generator[None, None, None]:
+        """Set configuration options temporarily inside a ``with`` statement.
```

Comment:
it's a given but it may be worth adding something a la "This configuration will only be effective inside this context manager!" - some users may think this could be used a permanent config setter

Reply:
Done in https://github.com/ESMValGroup/ESMValCore/pull/2778/commits/85043dfc5479564536e7530d4cc0e7edda0f3e28

Reply:
:beers: